session-manager/src/session.rs

```diff
@@ -43,27 +43,30 @@ pub struct Session {
 impl Session {
     /// Creates a new session from a server configuration and a list of
     /// principals.
-    pub fn new(config: rustls::ServerConfig, principals: &Vec<Principal>) -> Self {
-        let tls_session = ServerSession::new(&std::sync::Arc::new(config));
+    pub fn new(config: rustls::ServerConfig, principals: &Vec<Principal>) -> Result<Self, SessionManagerError> {
+        let mut tls_connection = ServerConnection::new(std::sync::Arc::new(config))?;
+        tls_connection.set_buffer_limit(Some(512 * 1024));
 
-        Session {
-            tls_session: tls_session,
+        Ok(Session {
+            tls_connection: tls_connection,
             principals: principals.to_vec(),
```

Comment:
Is it not better to take `principals` as a `Vec<Principal>` and avoid allocating here?

Reply:
What specific syntax are you talking about here? I'm not familiar with that pattern.